gcovr/gcov.py

```diff
@@ -483,15 +483,22 @@ def run_gcov_and_process_files(abs_filename, covdata, options, error, toerase, c
         # This lock is essential for parallel processing because without
         # this there can be name collisions for the generated output files.
         with locked_directory(chdir):
-            out, err = gcov_cmd.run_with_args(
+            gcov_process = gcov_cmd.run_with_args(
                 [
                     abs_filename,
                     *gcov_cmd.get_default_options(),
                     "--object-directory",
                     os.path.dirname(abs_filename),
                 ],
                 cwd=chdir,
-            ).communicate()
+            )
+            out, err = gcov_process.communicate()
+            if gcov_process.returncode < 0:
+                raise RuntimeError(
+                    f"GCOV returncode was {gcov_process.returncode} (exited by signal)."
+                )
+            elif gcov_process.returncode != 0:
+                raise RuntimeError(f"GCOV returncode was {gcov_process.returncode}.")
```

Comment:
Can this code be tested with some fault injection technique? I think something like `--gcov-executable 'bash -c "exit 1"'` could work for normal return codes, and `--gcov-executable 'bash -c "kill -9 \$\$"'` for the signal case (Unixoid systems only). But this doesn't have to be a full integration test, if this part of the logic is extracted into a separate function.

Reply:
I was also thinking about how to test this. A doctest is a good idea. Since we are testing with a installed Mays on Windows this should also work there.